src/cli/self_update.rs

```diff
@@ -390,29 +390,28 @@ fn check_existence_of_rustc_or_cargo_in_path(no_prompt: bool) -> Result<()> {
         err!("Otherwise you may have confusion unless you are careful with your PATH");
         err!("If you are sure that you want both rustup and your already installed Rust");
         err!("then please restart the installation and pass `-y' to bypass this check.");
-        Err("cannot install while Rust is installed".into())
-    } else {
-        Ok(())
+        non_fatal_error("cannot install while Rust is installed".into(), "\nContinue? (y/N)", false, no_prompt)?;
```

Comment:
I'm not convinced it's worth passing in this prompt, the sense of the prompt is essential to the logic of your `non_fatal_error()` function so I'd just have that string in there. Ditto for the default really.

Reply:
OK, makes sense. I guess i18n would be the most likely reason for changing the prompt, and that it doesn't seem like rustup does that at all right now.